src/shared/components/timer/timer.component.js

```diff
@@ -0,0 +1,123 @@
+import React, { PureComponent } from 'react';
+import moment from 'moment';
+import firebase from 'firebase';
+
+import { StartButton, SaveButton, Time, HistoryContainer } from './timer.styles';
+
+require('firebase/firestore');
+
+export class Timer extends PureComponent {
+  state = {
+    counting: false,
+    time: 0,
+    databaseTimes: [],
+  };
+
+  componentDidMount() {
+    this.db = firebase.firestore();
+    const times = [];
+    this.db
+      .collection('times')
+      .get()
+      .then(snapshot => {
+        snapshot.docs.map(doc => {
+          times.push(doc.data());
+        });
+        this.setState({ databaseTimes: [...times] });
+      });
+  }
+
+  componentWillUnmount() {
+    this.stopTimer();
+  }
+
+  startTimer = () => {
+    const referenceTime = moment.now();
+    this.timeInterval = setInterval(() => {
+      this.setState({ time: moment.now() - referenceTime });
+    });
+  };
+
+  stopTimer = () => clearInterval(this.timeInterval);
+
+  leftPad = (number, targetLength) => {
+    let output = number + '';
+    while (output.length < targetLength) {
+      output = '0' + output;
+    }
+    return output;
+  };
+
+  displayTime = time => {
+    this.minutes = Math.floor(time / 60000);
+    this.seconds = Math.floor((time % 60000) / 1000);
+    this.milliseconds = time % 1000;
+    return `${this.leftPad(this.minutes, 2)}:${this.leftPad(this.seconds, 2)}.${this.leftPad(this.milliseconds, 3)}`;
+  };
+
+  handleStartButton = () => {
+    this.state.counting ? this.stopTimer() : this.startTimer();
```

Comment:
Czy to aby napewno powinno tak działać?
Komponent powinien udostępniać `start` i `stop` poprzez refa, bo obstawiam, że nie bedziesz klikać buttona dla 1000 próbek, a raczej będzie się to dziać automatycznie :P
Zamiast udostępniania metod na zewnątrz można zrobić z tego komponentu hook'a - jak wspomniałem już chyba wcześniej. Zależy od Twoich wymagań.